wdl-ast/src/v1/document/expression/literal/string.rs

```diff
@@ -0,0 +1,330 @@
+//! Literal strings.
+
+pub mod inner;
+
+use grammar::v1::Rule;
+pub use inner::Inner;
+use pest::iterators::Pair;
+use wdl_grammar as grammar;
+use wdl_macros::check_node;
+
+/// A literal string.
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub enum String {
+    /// A single quoted string.
+    SingleQuoted(Inner),
+
+    /// A double quoted string.
+    DoubleQuoted(Inner),
```

Comment:
@peterhuene since Clay is out I'm asking you:
I would think that 95% of the time we don't care about the distinction between single quotes and double quotes. Doesn't this structure complicate usage by requiring handling of the enum variants? I don't know what an alternative would look like, but could this be handled differently? In a way that obfuscates the quote character for general usage (but of course we still need that information stored _somewhere_).

Does this make sense?

Reply:
I agree that it seems no longer necessary to make the distinction between single and double quoted strings in the AST; there's simply one "string literal" at this representation level.

In the `rowan`-based parser, there will only be a string literal "node" in the CST and the node would have child tokens for whatever opening/closing quote characters were in the parsed source.